include/fsm/fill_spill_merge.hpp

```diff
@@ -373,13 +375,15 @@ static void MoveWaterInDepHier(
     const auto lchild = this_dep.lchild;
     const auto rchild = this_dep.rchild;
 
+
     //Only if both children are full should their water make its way to this
     //parent
     if(lchild!=NO_VALUE
       && deps.at(lchild).water_vol==deps.at(lchild).dep_vol
       && deps.at(rchild).water_vol==deps.at(rchild).dep_vol
+      && deps.at(this_dep.dep_label).water_vol < FP_ERROR  
```

Comment:
Why?

Reply:
Because there were some cases where the parent depression (this_dep in this context) had already received the children's water at a different point in the code. We need to check whether it has already received the water, or it will count the children's water twice (maybe even more than twice? I'm not certain, but I do know that it was counting at least twice in some cases). 
Prior to counting its children's volume, a parent's water volume should be 0; cautiously I used < FP_ERROR instead of == 0.

Reply:
This check worries me because (maybe?) multiply flows lower down could pass water into the parent. I think it's cleaner to figure out how to structure the algorithm so the water only passes upwards once. I'll have to stare at this for a bit :-/

Reply:
I don't think they should since lower-down parents have also gone through this same check, but I'm all for a cleaner fix.